Userland/Libraries/LibLine/InternalFunctions.cpp

```diff
@@ -522,4 +526,91 @@ void Editor::uppercase_word()
     case_change_word(CaseChangeOp::Uppercase);
 }
 
+void Editor::edit_in_external_editor()
+{
+    const auto* editor_command = getenv("EDITOR");
+    if (!editor_command)
+        editor_command = m_configuration.m_default_text_editor.characters();
+
+    char file_path[] = "/tmp/line-XXXXXX";
+    auto fd = mkstemp(file_path);
+
+    if (fd < 0) {
+        perror("mktemp");
+        return;
+    }
+
+    {
+        auto* fp = fdopen(fd, "rw");
+        if (!fp) {
+            perror("fdopen");
+            return;
+        }
```

Comment:
This gunk makes me feel like we should have something in LibCore for creating a temp file.

Reply:
we have similar (but slightly different) gunk in https://github.com/SerenityOS/serenity/blob/master/Userland/Utilities/mktemp.cpp

Reply:
Definitely should, +1.